cmd/internal/ecs/add.go

```diff
@@ -0,0 +1,330 @@
+// errors should be wrapped to avoid showing usage.  (This is reversed
+// from the other command conventions, but there are relatively few
+// usage errors here.
+func RunAddWorkflow() error {
+	wf := &AddWorkflow{
+		currentState: initState,
+	}
+
+	keepGoing := true
+	var err error = nil
+	for keepGoing && err == nil {
+		keepGoing, err = wf.currentState(wf)
+	}
+	if err == nil {
+		telemetry.Success("Add to ECS")
+	} else if errors.Is(err, terminal.InterruptErr) {
+		printer.Infof("Interrupted!\n")
+		telemetry.WorkflowStep("Add to ECS", "User interrupted session")
+		return nil
+	} else if _, ok := err.(UsageError); ok {
+		telemetry.Error("Add to ECS", err)
+		return err
+	} else {
+		telemetry.Error("Add to ECS", err)
+		return cmderr.AkitaErr{Err: err}
+	}
+	return err
+}
+
+// Initial state: check if running interactively, if so then start
+// with collecting AWS profile.
+func initState(wf *AddWorkflow) (keepGoing bool, err error) {
+	reportStep("Start Add to ECS")
+
+	// Check if running interactively.
+	// TODO: I didn't see a way to do this from go-survey directly.
+	if !term.IsTerminal(int(os.Stdin.Fd())) {
+		return fillFromFlags(wf)
+	}
+
+	wf.currentState = getProfileState
+	return true, nil
+}
+
+// Ask the user to specify a profile; "" is fine to use the default profile.
+// TODO: it seems very difficult to present a list (which is what I was trying
+// to do orginally) because the SDK doesn't provide an API to do that, and
+// its config file parser is internal.
+func getProfileState(wf *AddWorkflow) (keepGoing bool, err error) {
+	reportStep("Get AWS Profile")
+
+	if awsProfileFlag != "" {
+		wf.awsProfile = awsProfileFlag
+		if err = wf.checkCredentials(); err != nil {
+			printer.Errorf("Could not find AWS credentials for profile %q. The error from the AWS library is shown below.\n")
+			return false, errors.Wrap(err, "Error loading credentials")
+		}
+
+		wf.currentState = getRegionState
+		return true, nil
+	}
+
+	// Use the existing value as the default in case we repeat this step
+	err = survey.AskOne(
+		&survey.Input{
+			Message: "Which of your AWS profiles should Akita use to configure ECS?",
+			Help:    "Enter the name of the AWS profile you use for configuring ECS, or leave blank to try the default profile. Akita needs this information to identify which AWS credentials to use.",
+			Default: wf.awsProfile,
+		},
+		&wf.awsProfile,
+	)
+	if err != nil {
+		return false, err
+	}
+
+	wf.awsCred = credentials.NewSharedCredentials(awsCredentialsFlag, wf.awsProfile)
+	if err = wf.checkCredentials(); err != nil {
+		printer.Errorf("Error from AWS library: %v\n", err)
+		printer.Errorf("Could not find AWS credentials for profile %q, please try again or hit Ctrl+C to exit.\n", wf.awsProfile)
+		wf.awsProfile = ""
+		return true, nil
+	}
+
+	// TODO: I don't think there's even a way to show the profile name here?
+	printer.Infof("Successfully loaded AWS credentials.\n")
```

Comment:
I was confused because I thought the default profile could be something other than "default", and so there was no way to load the actual value if we used "".

But, I was wrong on the first point, and the new version uses "default" as the default profile name instead of "", as the latter doesn't work with the new config anyway.